### PR TITLE
Update `tanzu context update  tanzu-active-resource` to support `--clustergroup` flag

### PR DESCRIPTION
**Pull request by anujc25**
### What this PR does / why we need it
- Update `tanzu context update  tanzu-active-resource` to support `--clustergroup` flag
- As part of this change, the context object within the `config-ng.yaml` file also saves `tanzuClusterGroupName` under the `additionalMetadata` section.
```
      additionalMetadata:
        tanzuClusterGroupName: clustergroupA
        tanzuOrgID: b1d48027-bb69-4a56-a5b8-e941ef29fa4b
        tanzuProjectName: projectA
        tanzuSpaceName: ""
```

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

```
## Set project as active
$ tz context update tanzu-active-resource one-tanzu-demo-1 --project "projectA"
$ tz context list
  NAME              ISACTIVE  TYPE        ENDPOINT                                                                         KUBECONFIGPATH             KUBECONTEXT                 PROJECT   SPACE  CLUSTERGROUP  
  one-tanzu-demo-1  false     tanzu       https://api.tanzu-dev.cloud.vmware.com/org/b1d48027-bb69-4a56-a5b8-e941ef29fa4b  /Users/anujc/.kube/config  tanzu-cli-one-tanzu-demo-1  projectA                       
  tkg-mc-1          true      kubernetes                                                                                   /Users/anujc/.kube/config  kind-mc-1                   n/a       n/a    n/a           
```
```
## Set space within a project as active
$ tz context update tanzu-active-resource one-tanzu-demo-1 --project "projectA" --space "spaceA"
$ tz context list
  NAME              ISACTIVE  TYPE        ENDPOINT                                                                         KUBECONFIGPATH             KUBECONTEXT                 PROJECT   SPACE   CLUSTERGROUP  
  one-tanzu-demo-1  false     tanzu       https://api.tanzu-dev.cloud.vmware.com/org/b1d48027-bb69-4a56-a5b8-e941ef29fa4b  /Users/anujc/.kube/config  tanzu-cli-one-tanzu-demo-1  projectA  spaceA                
  tkg-mc-1          true      kubernetes                                                                                   /Users/anujc/.kube/config  kind-mc-1                   n/a       n/a     n/a           
```
```
## Set clustergroup within a project as active
$ tz context update tanzu-active-resource one-tanzu-demo-1 --project "projectA" --clustergroup "clustergroupA"
$ tz context list
  NAME              ISACTIVE  TYPE        ENDPOINT                                                                         KUBECONFIGPATH             KUBECONTEXT                 PROJECT   SPACE  CLUSTERGROUP   
  one-tanzu-demo-1  false     tanzu       https://api.tanzu-dev.cloud.vmware.com/org/b1d48027-bb69-4a56-a5b8-e941ef29fa4b  /Users/anujc/.kube/config  tanzu-cli-one-tanzu-demo-1  projectA         clustergroupA  
  tkg-mc-1          true      kubernetes                                                                                   /Users/anujc/.kube/config  kind-mc-1                   n/a       n/a    n/a            
```

- Incorrect input tests:
```
## Try to set space and clustergroup both within a project as active
$ tz context update tanzu-active-resource one-tanzu-demo-1 --project "projectA" --space "spaceA"  --clustergroup "clustergroupA"
[x] : either space or clustergroup can be set as active resource. Please provide either --space or --clustergroup option

## Try to set clustergroup without a project
$ tz context update tanzu-active-resource one-tanzu-demo-1 --clustergroup "clustergroupA"
[x] : clustergroup cannot be set without project name. Please provide project name also using --project option

## Try to set space without a project
$ tz context update tanzu-active-resource one-tanzu-demo-1 --space "spaceA"
[x] : space cannot be set without project name. Please provide project name also using --project option
```

### Release note

```release-note
Support configuring ClusterGroup with the `tanzu context update  tanzu-active-resource` command using `--clustergroup` flag
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,6 +6,8 @@ replace cloud.google.com/go => cloud.google.com/go v0.102.1
 
 replace github.com/vmware-tanzu/tanzu-cli/test/e2e/framework => ./test/e2e/framework
 
+replace github.com/vmware-tanzu/tanzu-plugin-runtime => github.com/anujc25/tanzu-plugin-runtime v0.0.0-20231215153217-450decceb82b
```
Comment (anujc25) [inline go.mod:9]: Need to be updated once https://github.com/vmware-tanzu/tanzu-plugin-runtime/pull/142 is merged.

**Review by marckhouzam — APPROVED** [review]
I assume any improvement to `context list` will be done in a separate PR?

LGTM

**Comment by anujc25** [comment]
> I assume any improvement to `context list` will be done in a separate PR?
> 
> LGTM

Correct. UX changes around the `tanzu context list` will be part of a separate PR.

File: pkg/command/context.go [hunk 2]
```diff
@@ -1408,6 +1423,7 @@ func setTanzuCtxActiveResource(_ *cobra.Command, args []string) error {
 	}
 	ctx.AdditionalMetadata[config.ProjectNameKey] = projectStr
 	ctx.AdditionalMetadata[config.SpaceNameKey] = spaceStr
+	ctx.AdditionalMetadata[config.ClusterGroupNameKey] = clustergroupStr
```
Comment (prkalle) [inline pkg/command/context.go:1426]: I think we should also update the TanzuContextKubeconfig at L1431 to reflect the clustergroup endpoint.

**Review by prkalle — APPROVED** [review]
lgtm. Thanks for the changes!

**Pull request merged**